tests/test_framework/mod.rs

```diff
@@ -142,7 +142,7 @@ impl TestFramework {
         );
         let directory_server_instance_clone = directory_server_instance.clone();
         thread::spawn(move || {
-            start_directory_server(directory_server_instance_clone).expect("TODO: panic message");
+            start_directory_server(directory_server_instance_clone).unwrap();
```

Comment:
```rust
thread::spawn(move || {
    if let Err(e) = start_directory_server(directory_server_instance_clone) {
        return; 
    }
});
```

Reply:
We can add a log

Reply:
We can either add a logic to change the directory server after a certain set of retries or abort the maker instantiation, in case the directory server is not responding after certain hits.

Reply:
This is were error handling shines, we can return result from thread spawn, and based on that can have a revival mechanism for directory. Can be taken up as a separate PR, and the error propagation will help us to know which solution to be taken.

Reply:
It's in the test. Unwrap is fine. we will get the error message at the end of the test run. A more elaborate thread and error handling could be added. But not touching it in this PR.